docs/core/transactions/confirmation.md

```diff
@@ -59,61 +70,65 @@ hashes to build a trusted clock. The “history” part of the name comes from t
 are related to how RPC nodes and validators detect and handle **expired**
 transactions. A solid understanding of how transaction expiration works should
 help you diagnose the bulk of your transaction confirmation issues.
 
-## How does transaction expiration work?
+### How does transaction expiration work?
 
 Each transaction includes a “recent blockhash” which is used as a PoH clock
-timestamp and expires when that blockhash is no longer “recent” enough. More
-concretely, Solana validators look up the corresponding slot number for each
-transaction’s blockhash that they wish to process in a block. If the validator
-[can’t find a slot number for the blockhash](https://github.com/solana-labs/solana/blob/9488a73f5252ad0d7ea830a0b456d9aa4bfbb7c1/runtime/src/bank.rs#L3687)
-or if the looked up slot number is more than 151 slots lower than the slot
-number of the block being processed, the transaction will be rejected.
-
-Slots are configured to last about
-[400ms](https://github.com/solana-labs/solana/blob/47b938e617b77eb3fc171f19aae62222503098d7/sdk/program/src/clock.rs#L12)
-but often fluctuate between 400ms and 600ms, so a given blockhash can only be
-used by transactions for about 60 to 90 seconds.
-
-Transaction has expired pseudocode:
-`currentBankSlot > slotForTxRecentBlockhash + 151`
-
-Transaction not expired pseudocode:
-`currentBankSlot - slotForTxRecentBlockhash < 152`
+timestamp and expires when that blockhash is no longer “recent enough”.
+
+As each block is finalized (i.e. the maximum tick height
+[is reached](https://github.com/anza-xyz/agave/blob/0588ecc6121ba026c65600d117066dbdfaf63444/runtime/src/bank.rs#L3269-L3271),
+reaching the "block boundary"), the final hash of the block is added to the
+`BlockhashQueue` which stores a maximum of the
+[300 most recent blockhashes](https://github.com/anza-xyz/agave/blob/e0b0bcc80380da34bb63364cc393801af1e1057f/sdk/program/src/clock.rs#L123-L126).
+During transaction processing, Solana Validators will check if each
+transaction's recent blockhash is recorded within the
+[150 most recent blockhashes](https://github.com/anza-xyz/agave/blob/cb2fd2b632f16a43eff0c27af7458e4e97512e31/sdk/program/src/clock.rs#L129-L131)
+(aka the max processing age) stored within the `BlockhashQueue`. If the
+transaction's recent blockhash is
+[older than this](https://github.com/anza-xyz/agave/blob/cb2fd2b632f16a43eff0c27af7458e4e97512e31/runtime/src/bank.rs#L3570-L3571)
+max processing age, the transaction is not processed.
+
+Since [slots](/docs/terminology.md#slot) (aka the time period a validator can
+produce a block) are configured to last about
+[400ms](https://github.com/anza-xyz/agave/blob/cb2fd2b632f16a43eff0c27af7458e4e97512e31/sdk/program/src/clock.rs#L107-L109),
+but may fluctuate between 400ms and 600ms, a given blockhash can only be used by
+transactions for about 60 to 90 seconds before it will be considered expired by
+the runtime.
```

Comment:
It might be worth also mentioning that blocks can be skipped, which would increase the lifetime of a blockhash

Reply:
A good callout. I will create a new PR to add some info on this :)